courseScheduleII.js

```diff
@@ -0,0 +1,113 @@
+/*
+For this approach, we can represent the input as a dependency graph where in order to take a course, 
+you have to complete a prereq, which is the same as two nodes connected by an edge.
+
+Since we are trying to find a valid ordering for all these courses that respects our prereqs,
+we can approach this problem with a topological sort algorithm where if there happens to be an
+edge between a prereq and a class, it means that the prereq appears before the class in a linear ordering fashion.
+
+A class that has a prereq before would be an indegree relationship and we can the can use our topological sort algorithm to track
+the number of indegrees. We can create a hashmap, in this case a JavsScript object, that maps all the classes
+to its prereqs and decrease the number of indegrees once we process that class.
+
+That way we're left with 0 indegree classes in a valid order.
+*/
+
```

Comment:
My latest changes removed your previous comment once I pushed my code. But if you look at the previous commit, you can see your comment. I basically updated my answer for the lines above.